ntrack/listener.go

```diff
@@ -0,0 +1,92 @@
+package ntrack
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync/atomic"
+
+	"github.com/pkg/errors"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+type trackingListener struct {
+	net.Listener
+	stats *Stats
+}
+
+func NewInstrumentedListener(lis net.Listener) (net.Listener, *Stats) {
+	listenerStats := &Stats{}
+	listenerStats.init()
+
+	return &trackingListener{
+		Listener: lis,
+		stats:    listenerStats,
+	}, listenerStats
+}
+
+func (tl *trackingListener) Accept() (net.Conn, error) {
+	conn, err := tl.Listener.Accept()
+	stats.RecordWithTags(context.TODO(), []tag.Mutator{tag.Upsert(tl.stats.TagSuccess, fmt.Sprintf("%v", err == nil))}, tl.stats.ListenerAccepted.M(1))
```

Comment:
Perhaps you might want to have two tag keys
status --> whose values can either be "OK" and "ERROR"
error --> whose value is set only if key "status" is set and its value is the full error encountered.

This will aid a lot with grouping errors, creating alerts on various errors and determination of error/success rates.

Reply:
Interesting. having arbitrary strings end up as tags is a little icky, but I suspect there's only a small number of them you'd possibly encounter from the stdlib. 

I can see how this would be very useful, although my instinct would be to check for the actual error strings in logs/traces.

Reply:
I also don't like arbitrary strings as tags (if strings have values like port number, ip address, stacks, etc the cardinality can blow up). A compromise is to add it here but not add it by default in the respective view and have some way (either code or doc) to add it explicitly if one wants it. 

I would recommend pre-defining two mutator slices and select the one to be used according to the value of err instead of creating them on every call.